File: 0263-ugly-number/0263-ugly-number.py
import math, sys
sys.setrecursionlimit(100000)
class Solution:
    def isUgly(self, n: int) -> bool:
        if n < 1:
            return False
        if n in (1, 2, 3, 5):
            return True
        
        if n % 2 == 0:
            return self.isUgly(n // 2)
        elif n % 3 == 0:
            return self.isUgly(n // 3)
        elif n % 5 == 0:
            return self.isUgly(n // 5)
        return False
            

File: 0263-ugly-number/0263-ugly-number.rs
impl Solution {
    pub fn is_ugly(n: i32) -> bool {
        if n < 1 {
            return false;
        }
        
        if n == 1 || n == 2 || n == 3 || n == 5 {
            return true;
        }
        
        if n % 2 == 0 {
            return Solution::is_ugly(n / 2);
        } else if n % 3 == 0 {
            return Solution::is_ugly(n / 3);
        } else if n % 5 == 0 {
            return Solution::is_ugly(n / 5);
        }
        
        return false
    }
}